// @radzen/angular/dist/boolean-form-field.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var form_field_1 = require("./form-field");
var BooleanFormFieldComponent = /** @class */ (function (_super) {
    __extends(BooleanFormFieldComponent, _super);
    function BooleanFormFieldComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    BooleanFormFieldComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-boolean-form-field',
                    template: "\n    <p-checkbox *ngIf=\"required\"\n      [formControl]=\"parent.form.controls[property]\"\n      [inputId]=\"property\"\n      [binary]=\"true\">\n    </p-checkbox>\n    <p-triStateCheckbox *ngIf=\"!required\"\n      [inputId]=\"property\"\n      [formControl]=\"parent.form.controls[property]\">\n    </p-triStateCheckbox>\n    <div class=\"md-inputfield\"\n      *ngIf=\"(parent.form.controls[property].touched || parent.submitted) && parent.form.controls[property].errors?.required\">\n      <div class=\"ui-message ui-messages-error ui-corner-all\">\n        {{ title }} {{ requiredText }}\n      </div>\n    </div>\n  ",
                    providers: [
                        {
                            provide: form_field_1.FormFieldComponent,
                            useExisting: core_1.forwardRef(function () { return BooleanFormFieldComponent; })
                        }
                    ]
                },] },
    ];
    /** @nocollapse */
    BooleanFormFieldComponent.ctorParameters = function () { return []; };
    return BooleanFormFieldComponent;
}(form_field_1.FormFieldComponent));
exports.BooleanFormFieldComponent = BooleanFormFieldComponent;
