// @radzen/angular/dist/button.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var ButtonComponent = /** @class */ (function () {
    function ButtonComponent() {
        this.text = '';
        this.size = 'md';
        this.buttonStyle = 'primary';
        this.click = new core_1.EventEmitter();
        this.style = {};
        this.disabled = false;
        this.buttonType = 'button';
    }
    ButtonComponent.prototype.onClick = function (event) {
        event.stopPropagation();
        this.click.next();
    };
    ButtonComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-button',
                    template: "\n    <button\n      [ngStyle]=\"style\"\n      class=\"ui-button ui-button-{{size}} ui-widget ui-state-default ui-corner-all btn-{{buttonStyle}}\"\n      [type]=\"buttonType\"\n      [disabled]=\"disabled\"\n      [ngClass]=\"{\n        'ui-state-disabled': disabled,\n        'ui-button-text-icon-left': icon && text,\n        'ui-button-icon-only': icon && !text\n      }\"\n      (click)=\"onClick($event)\"\n    >\n      <i *ngIf=\"icon\" class=\"ui-button-icon-left fa\">{{ icon }}</i>\n      <span *ngIf=\"text\" class=\"ui-button-text\">{{ text }}</span>\n    </button>\n  "
                },] },
    ];
    /** @nocollapse */
    ButtonComponent.ctorParameters = function () { return []; };
    ButtonComponent.propDecorators = {
        "text": [{ type: core_1.Input },],
        "icon": [{ type: core_1.Input },],
        "size": [{ type: core_1.Input },],
        "buttonStyle": [{ type: core_1.Input },],
        "click": [{ type: core_1.Output },],
        "style": [{ type: core_1.Input },],
        "disabled": [{ type: core_1.Input },],
        "buttonType": [{ type: core_1.Input },],
    };
    return ButtonComponent;
}());
exports.ButtonComponent = ButtonComponent;
var ButtonModule = /** @class */ (function () {
    function ButtonModule() {
    }
    ButtonModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule],
                    declarations: [ButtonComponent],
                    exports: [ButtonComponent]
                },] },
    ];
    /** @nocollapse */
    ButtonModule.ctorParameters = function () { return []; };
    return ButtonModule;
}());
exports.ButtonModule = ButtonModule;
