// @radzen/angular/dist/card.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var CardComponent = /** @class */ (function () {
    function CardComponent() {
        this.style = {};
    }
    CardComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-card',
                    template: "\n    <div class=\"card\" [ngStyle]=\"style\">\n      <ng-content></ng-content>\n    </div>\n  "
                },] },
    ];
    /** @nocollapse */
    CardComponent.ctorParameters = function () { return []; };
    CardComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
    };
    return CardComponent;
}());
exports.CardComponent = CardComponent;
var CardModule = /** @class */ (function () {
    function CardModule() {
    }
    CardModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule],
                    declarations: [CardComponent],
                    exports: [CardComponent]
                },] },
    ];
    /** @nocollapse */
    CardModule.ctorParameters = function () { return []; };
    return CardModule;
}());
exports.CardModule = CardModule;
