// @radzen/angular/dist/checkboxlist.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var forms_1 = require("@angular/forms");
var checkbox_1 = require("primeng/checkbox");
var form_component_1 = require("./form-component");
var options_component_1 = require("./options-component");
var CheckBoxListOptionComponent = /** @class */ (function () {
    function CheckBoxListOptionComponent() {
    }
    CheckBoxListOptionComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-checkboxlist-option',
                    template: ''
                },] },
    ];
    /** @nocollapse */
    CheckBoxListOptionComponent.ctorParameters = function () { return []; };
    CheckBoxListOptionComponent.propDecorators = {
        "text": [{ type: core_1.Input },],
        "value": [{ type: core_1.Input },],
    };
    return CheckBoxListOptionComponent;
}());
exports.CheckBoxListOptionComponent = CheckBoxListOptionComponent;
var CheckBoxListComponent = /** @class */ (function (_super) {
    __extends(CheckBoxListComponent, _super);
    function CheckBoxListComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.style = {};
        _this.orientation = 'horizontal';
        return _this;
    }
    CheckBoxListComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-checkboxlist',
                    template: "\n    <div\n      [id]=\"name\"\n      [ngStyle]=\"style\"\n      [ngClass]=\"{\n        'checkbox-list-horizontal': orientation == 'horizontal',\n        'checkbox-list-vertical': orientation == 'vertical'\n      }\"\n    >\n      <p-checkbox\n        *ngFor=\"let option of options\"\n        [value]=\"option.value\"\n        [label]=\"option.label\"\n        [(ngModel)]=\"value\"\n        [name]=\"name\"\n        [disabled]=\"disabled\"\n        (blur)=\"onTouched()\"\n        (change)=\"$event.stopPropagation()\"\n      >\n      </p-checkbox>\n    </div>\n  ",
                    providers: [
                        {
                            provide: forms_1.NG_VALUE_ACCESSOR,
                            useExisting: core_1.forwardRef(function () { return CheckBoxListComponent; }),
                            multi: true
                        },
                        {
                            provide: form_component_1.FormComponent,
                            useExisting: core_1.forwardRef(function () { return CheckBoxListComponent; })
                        }
                    ]
                },] },
    ];
    /** @nocollapse */
    CheckBoxListComponent.ctorParameters = function () { return []; };
    CheckBoxListComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "orientation": [{ type: core_1.Input },],
        "children": [{ type: core_1.ContentChildren, args: [CheckBoxListOptionComponent,] },],
    };
    return CheckBoxListComponent;
}(options_component_1.OptionsComponent));
exports.CheckBoxListComponent = CheckBoxListComponent;
var CheckBoxListModule = /** @class */ (function () {
    function CheckBoxListModule() {
    }
    CheckBoxListModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, checkbox_1.CheckboxModule, forms_1.FormsModule],
                    declarations: [CheckBoxListOptionComponent, CheckBoxListComponent],
                    exports: [CheckBoxListOptionComponent, CheckBoxListComponent]
                },] },
    ];
    /** @nocollapse */
    CheckBoxListModule.ctorParameters = function () { return []; };
    return CheckBoxListModule;
}());
exports.CheckBoxListModule = CheckBoxListModule;
