// @radzen/angular/dist/checkbox.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var forms_1 = require("@angular/forms");
var tristatecheckbox_1 = require("primeng/tristatecheckbox");
var checkbox_1 = require("primeng/checkbox");
var form_component_1 = require("./form-component");
var CheckBoxComponent = /** @class */ (function (_super) {
    __extends(CheckBoxComponent, _super);
    function CheckBoxComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.style = {};
        _this.triState = false;
        return _this;
    }
    CheckBoxComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-checkbox',
                    template: "\n  <p-checkbox *ngIf=\"!triState\"\n    [style]=\"style\"\n    [inputId]=\"name\"\n    [(ngModel)]=\"value\"\n    [binary]=\"true\"\n    [disabled]=\"disabled\"\n    (blur)=\"onTouched()\"\n    (change)=\"$event.stopPropagation()\">\n  </p-checkbox>\n  <p-triStateCheckbox *ngIf=\"triState\"\n    [style]=\"style\"\n    [(ngModel)]=\"value\"\n    [inputId]=\"name\"\n    [disabled]=\"disabled\"\n    (blur)=\"onTouched()\"\n    (change)=\"$event.stopPropagation()\">\n  </p-triStateCheckbox>",
                    providers: [
                        {
                            provide: forms_1.NG_VALUE_ACCESSOR,
                            useExisting: core_1.forwardRef(function () { return CheckBoxComponent; }),
                            multi: true
                        },
                        {
                            provide: form_component_1.FormComponent,
                            useExisting: core_1.forwardRef(function () { return CheckBoxComponent; })
                        }
                    ]
                },] },
    ];
    /** @nocollapse */
    CheckBoxComponent.ctorParameters = function () { return []; };
    CheckBoxComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "triState": [{ type: core_1.Input },],
    };
    return CheckBoxComponent;
}(form_component_1.FormComponent));
exports.CheckBoxComponent = CheckBoxComponent;
var CheckBoxModule = /** @class */ (function () {
    function CheckBoxModule() {
    }
    CheckBoxModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [
                        common_1.CommonModule,
                        forms_1.FormsModule,
                        checkbox_1.CheckboxModule,
                        tristatecheckbox_1.TriStateCheckboxModule
                    ],
                    declarations: [CheckBoxComponent],
                    exports: [CheckBoxComponent]
                },] },
    ];
    /** @nocollapse */
    CheckBoxModule.ctorParameters = function () { return []; };
    return CheckBoxModule;
}());
exports.CheckBoxModule = CheckBoxModule;
