// @radzen/angular/dist/content-container.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var document_token_1 = require("./document.token");
var content_1 = require("./content");
var ContentContainerComponent = /** @class */ (function () {
    function ContentContainerComponent(containerRef, contentService, documentToken) {
        var _this = this;
        this.containerRef = containerRef;
        this.contentService = contentService;
        this.documentToken = documentToken;
        this.subscription = contentService
            .afterContentRegistered()
            .subscribe(function (content) {
            if (_this.documentToken == null) {
                if (content.container == _this.name) {
                    var nativeElement = _this.containerRef.element.nativeElement;
                    nativeElement.replaceChild(content.elementRef.nativeElement, nativeElement.firstChild);
                }
            }
        });
    }
    ContentContainerComponent.prototype.ngOnDestroy = function () {
        this.subscription.unsubscribe();
    };
    ContentContainerComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-content-container',
                    template: "<ng-content></ng-content>"
                },] },
    ];
    /** @nocollapse */
    ContentContainerComponent.ctorParameters = function () { return [
        { type: core_1.ViewContainerRef, },
        { type: content_1.ContentService, },
        { type: undefined, decorators: [{ type: core_1.Optional }, { type: core_1.Inject, args: [document_token_1.DOCUMENT_TOKEN,] },] },
    ]; };
    ContentContainerComponent.propDecorators = {
        "name": [{ type: core_1.Input },],
    };
    return ContentContainerComponent;
}());
exports.ContentContainerComponent = ContentContainerComponent;
var ContentContainerModule = /** @class */ (function () {
    function ContentContainerModule() {
    }
    ContentContainerModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, content_1.ContentModule],
                    declarations: [ContentContainerComponent],
                    exports: [ContentContainerComponent]
                },] },
    ];
    /** @nocollapse */
    ContentContainerModule.ctorParameters = function () { return []; };
    return ContentContainerModule;
}());
exports.ContentContainerModule = ContentContainerModule;
