// @radzen/angular/dist/footer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var transition_1 = require("./transition");
var layout_1 = require("./layout");
var FooterComponent = /** @class */ (function () {
    function FooterComponent(layoutService) {
        this.layoutService = layoutService;
        this.style = {};
        this.fixed = true;
    }
    FooterComponent.prototype.ngAfterViewInit = function () {
        this.layoutService.footerHeight.next(this.fixed ? this.footer.nativeElement.offsetHeight : 0);
    };
    FooterComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-footer',
                    template: "\n  <div class=\"footer\" [ngStyle]=\"style\" [ngClass]=\"{fixed:fixed}\" #footer rz-offset-by-sidebar\n      [@open]=\"{value: offset.sidebarOpen && offset.sidebarStretch ? 'visible': 'hidden', params: {marginLeft: offset.sidebarWidth}}\">\n    <ng-content></ng-content>\n  </div>\n  ",
                    animations: [transition_1.sidebarAnimation]
                },] },
    ];
    /** @nocollapse */
    FooterComponent.ctorParameters = function () { return [
        { type: layout_1.LayoutService, },
    ]; };
    FooterComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "fixed": [{ type: core_1.Input },],
        "footer": [{ type: core_1.ViewChild, args: ['footer',] },],
        "offset": [{ type: core_1.ViewChild, args: [layout_1.OffsetBySidebarDirective,] },],
    };
    return FooterComponent;
}());
exports.FooterComponent = FooterComponent;
var FooterModule = /** @class */ (function () {
    function FooterModule() {
    }
    FooterModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, layout_1.LayoutModule],
                    declarations: [FooterComponent],
                    exports: [FooterComponent]
                },] },
    ];
    /** @nocollapse */
    FooterModule.ctorParameters = function () { return []; };
    return FooterModule;
}());
exports.FooterModule = FooterModule;
