// @radzen/angular/dist/gauge.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var ngx_charts_1 = require("@swimlane/ngx-charts");
var shared_1 = require("./shared");
var GaugeComponent = /** @class */ (function () {
    function GaugeComponent() {
        this.style = {};
        this.value = 0;
        this.min = 0;
        this.max = 100;
        this.angleSpan = 240;
        this.startAngle = -120;
        this.showAxis = false;
        this.color = '#479cc8';
        this.smallSegments = 2;
        this.largeSegments = 5;
    }
    GaugeComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-gauge',
                    template: "\n    <div class=\"ui-gauge\" [ngStyle]=\"style\">\n      <ngx-charts-gauge\n        [results]=\"value != null ? [{name:'', value: value}] : []\"\n        [min]=\"min\"\n        [units]=\"unit\"\n        [scheme]=\"{domain:[color]}\"\n        [textValue]=\"text\"\n        [angleSpan]=\"angleSpan\"\n        [startAngle]=\"startAngle\"\n        [showAxis]=\"showAxis\"\n        [animations]=\"false\"\n        [bigSegments]=\"largeSegments\"\n        [smallSegments]=\"smallSegments\"\n        [margin]=\"showAxis ? null : [2,2,2,2]\"\n        [max]=\"max\">\n      </ngx-charts-gauge>\n    </div>\n  "
                },] },
    ];
    /** @nocollapse */
    GaugeComponent.ctorParameters = function () { return []; };
    GaugeComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "value": [{ type: core_1.Input },],
        "min": [{ type: core_1.Input },],
        "max": [{ type: core_1.Input },],
        "unit": [{ type: core_1.Input },],
        "angleSpan": [{ type: core_1.Input },],
        "startAngle": [{ type: core_1.Input },],
        "showAxis": [{ type: core_1.Input },],
        "text": [{ type: core_1.Input },],
        "color": [{ type: core_1.Input },],
        "smallSegments": [{ type: core_1.Input },],
        "largeSegments": [{ type: core_1.Input },],
    };
    return GaugeComponent;
}());
exports.GaugeComponent = GaugeComponent;
var GaugeModule = /** @class */ (function () {
    function GaugeModule() {
    }
    GaugeModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, ngx_charts_1.NgxChartsModule, shared_1.SharedModule],
                    declarations: [GaugeComponent],
                    exports: [GaugeComponent]
                },] },
    ];
    /** @nocollapse */
    GaugeModule.ctorParameters = function () { return []; };
    return GaugeModule;
}());
exports.GaugeModule = GaugeModule;
