// @radzen/angular/dist/gravatar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var md5 = require('blueimp-md5');
var GravatarComponent = /** @class */ (function () {
    function GravatarComponent() {
        this.style = {};
    }
    Object.defineProperty(GravatarComponent.prototype, "email", {
        set: function (email) {
            if (email) {
                var style = 'retro';
                var _a = this.style.width, width = _a === void 0 ? 36 : _a;
                email = email.toLowerCase().trim();
                this.url = "https://secure.gravatar.com/avatar/" + md5(email) + "?d=" + style + "&s=" + width;
            }
        },
        enumerable: true,
        configurable: true
    });
    GravatarComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-gravatar',
                    template: "\n    <img [src]=\"url\" class=\"gravatar\" [ngStyle]=\"style\">\n  "
                },] },
    ];
    /** @nocollapse */
    GravatarComponent.ctorParameters = function () { return []; };
    GravatarComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "email": [{ type: core_1.Input },],
    };
    return GravatarComponent;
}());
exports.GravatarComponent = GravatarComponent;
var GravatarModule = /** @class */ (function () {
    function GravatarModule() {
    }
    GravatarModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule],
                    declarations: [GravatarComponent],
                    exports: [GravatarComponent]
                },] },
    ];
    /** @nocollapse */
    GravatarModule.ctorParameters = function () { return []; };
    return GravatarModule;
}());
exports.GravatarModule = GravatarModule;
