// @radzen/angular/dist/heading.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var HeadingComponent = /** @class */ (function () {
    function HeadingComponent() {
        this.style = {};
    }
    HeadingComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-heading',
                    template: "\n    <h1 class=\"rz-heading\" *ngIf=\"size == 'H1'\" [ngStyle]=\"style\">{{text}}</h1>\n    <h2 class=\"rz-heading\" *ngIf=\"size == 'H2'\" [ngStyle]=\"style\">{{text}}</h2>\n    <h3 class=\"rz-heading\" *ngIf=\"size == 'H3'\" [ngStyle]=\"style\">{{text}}</h3>\n    <h4 class=\"rz-heading\" *ngIf=\"size == 'H4'\" [ngStyle]=\"style\">{{text}}</h4>\n    <h5 class=\"rz-heading\" *ngIf=\"size == 'H5'\" [ngStyle]=\"style\">{{text}}</h5>\n    <h6 class=\"rz-heading\" *ngIf=\"size == 'H6'\" [ngStyle]=\"style\">{{text}}</h6>\n  "
                },] },
    ];
    /** @nocollapse */
    HeadingComponent.ctorParameters = function () { return []; };
    HeadingComponent.propDecorators = {
        "text": [{ type: core_1.Input },],
        "size": [{ type: core_1.Input },],
        "style": [{ type: core_1.Input },],
    };
    return HeadingComponent;
}());
exports.HeadingComponent = HeadingComponent;
var HeadingModule = /** @class */ (function () {
    function HeadingModule() {
    }
    HeadingModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule],
                    declarations: [HeadingComponent],
                    exports: [HeadingComponent]
                },] },
    ];
    /** @nocollapse */
    HeadingModule.ctorParameters = function () { return []; };
    return HeadingModule;
}());
exports.HeadingModule = HeadingModule;
