// @radzen/angular/dist/header.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var transition_1 = require("./transition");
var layout_1 = require("./layout");
var HeaderComponent = /** @class */ (function () {
    function HeaderComponent(layoutService) {
        this.layoutService = layoutService;
        this.fixed = true;
        this.style = {};
    }
    HeaderComponent.prototype.ngDoCheck = function () {
        this.layoutService.headerHeight.next(this.fixed ? this.header.nativeElement.offsetHeight : 0);
    };
    HeaderComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-header',
                    template: "\n  <div #header class=\"header\" rz-offset-by-sidebar [ngStyle]=\"style\" [ngClass]=\"{fixed:fixed}\"\n    [@open]=\"{value: offset.sidebarOpen && offset.sidebarStretch ? 'visible': 'hidden', params: {marginLeft: offset.sidebarWidth}}\">\n    <ng-content></ng-content>\n  </div>\n  ",
                    animations: [transition_1.sidebarAnimation]
                },] },
    ];
    /** @nocollapse */
    HeaderComponent.ctorParameters = function () { return [
        { type: layout_1.LayoutService, },
    ]; };
    HeaderComponent.propDecorators = {
        "fixed": [{ type: core_1.Input },],
        "style": [{ type: core_1.Input },],
        "header": [{ type: core_1.ViewChild, args: ['header',] },],
        "offset": [{ type: core_1.ViewChild, args: [layout_1.OffsetBySidebarDirective,] },],
    };
    return HeaderComponent;
}());
exports.HeaderComponent = HeaderComponent;
var HeaderModule = /** @class */ (function () {
    function HeaderModule() {
    }
    HeaderModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, layout_1.LayoutModule],
                    declarations: [HeaderComponent],
                    exports: [HeaderComponent]
                },] },
    ];
    /** @nocollapse */
    HeaderModule.ctorParameters = function () { return []; };
    return HeaderModule;
}());
exports.HeaderModule = HeaderModule;
