// @radzen/angular/dist/image.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var shared_1 = require("./shared");
var ImageComponent = /** @class */ (function () {
    function ImageComponent() {
        this.style = {};
        this.click = new core_1.EventEmitter();
    }
    ImageComponent.prototype.onClick = function (event) {
        event.stopPropagation();
        this.click.next();
    };
    ImageComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-image',
                    template: "\n    <img (click)=\"onClick($event)\" [attr.src]=\"path | safe\" [ngStyle]=\"style\" />\n  "
                },] },
    ];
    /** @nocollapse */
    ImageComponent.ctorParameters = function () { return []; };
    ImageComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "path": [{ type: core_1.Input },],
        "click": [{ type: core_1.Output },],
    };
    return ImageComponent;
}());
exports.ImageComponent = ImageComponent;
var ImageModule = /** @class */ (function () {
    function ImageModule() {
    }
    ImageModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, shared_1.SharedModule],
                    declarations: [ImageComponent],
                    exports: [ImageComponent]
                },] },
    ];
    /** @nocollapse */
    ImageModule.ctorParameters = function () { return []; };
    return ImageModule;
}());
exports.ImageModule = ImageModule;
