// @radzen/angular/dist/label.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var LabelComponent = /** @class */ (function () {
    function LabelComponent() {
        this.text = '';
        this.style = {};
    }
    LabelComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-label',
                    template: "<label [ngStyle]=\"style\" [attr.for]=\"component\">{{ text }}</label>"
                },] },
    ];
    /** @nocollapse */
    LabelComponent.ctorParameters = function () { return []; };
    LabelComponent.propDecorators = {
        "text": [{ type: core_1.Input },],
        "component": [{ type: core_1.Input },],
        "style": [{ type: core_1.Input },],
    };
    return LabelComponent;
}());
exports.LabelComponent = LabelComponent;
var LabelModule = /** @class */ (function () {
    function LabelModule() {
    }
    LabelModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule],
                    declarations: [LabelComponent],
                    exports: [LabelComponent]
                },] },
    ];
    /** @nocollapse */
    LabelModule.ctorParameters = function () { return []; };
    return LabelModule;
}());
exports.LabelModule = LabelModule;
