// @radzen/angular/dist/link.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var router_1 = require("@angular/router");
var LinkComponent = /** @class */ (function () {
    function LinkComponent(differs) {
        this.differs = differs;
        this.style = {};
    }
    Object.defineProperty(LinkComponent.prototype, "path", {
        set: function (value) {
            this.route = null;
            this.href = null;
            if (value) {
                if (value.startsWith('/') || value.includes('://')) {
                    this.href = value;
                }
                else {
                    this.route = "/" + value;
                }
            }
        },
        enumerable: true,
        configurable: true
    });
    LinkComponent.prototype.ngDoCheck = function () {
        if (!this.differ && this.parameters) {
            this.differ = this.differs.find(this.parameters).create();
        }
        if (this.differ) {
            var changes = this.differ.diff(this.parameters);
            if (changes && this.route) {
                var _a = this.route.split('/'), _ = _a[0], path = _a[1];
                var segments = this.parameters.map(function (parameter) { return parameter.value; });
                this.route = "/" + [path].concat(segments).join('/');
            }
        }
    };
    LinkComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-link',
                    template: "\n    <a *ngIf=\"route\" [routerLink]=\"route\" class=\"link\" [ngStyle]=\"style\">\n      <i class=\"material-icons link-icon\" *ngIf=\"icon\">{{icon}}</i><span class=\"link-text\">{{text}}</span>\n    </a>\n    <a *ngIf=\"!route\" [attr.href]=\"href\" class=\"link\" [ngStyle]=\"style\">\n      <i class=\"material-icons link-icon\" *ngIf=\"icon\">{{icon}}</i><span class=\"link-text\">{{text}}</span>\n    </a>\n  "
                },] },
    ];
    /** @nocollapse */
    LinkComponent.ctorParameters = function () { return [
        { type: core_1.IterableDiffers, },
    ]; };
    LinkComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "text": [{ type: core_1.Input },],
        "icon": [{ type: core_1.Input },],
        "parameters": [{ type: core_1.Input },],
        "path": [{ type: core_1.Input },],
    };
    return LinkComponent;
}());
exports.LinkComponent = LinkComponent;
var LinkModule = /** @class */ (function () {
    function LinkModule() {
    }
    LinkModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, router_1.RouterModule],
                    declarations: [LinkComponent],
                    exports: [LinkComponent]
                },] },
    ];
    /** @nocollapse */
    LinkModule.ctorParameters = function () { return []; };
    return LinkModule;
}());
exports.LinkModule = LinkModule;
