// @radzen/angular/dist/lookup-popup-form-field.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var get = require("lodash/get");
var form_field_1 = require("./form-field");
var grid_1 = require("./grid");
var LookupPopupFormFieldComponent = /** @class */ (function (_super) {
    __extends(LookupPopupFormFieldComponent, _super);
    function LookupPopupFormFieldComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.emptyText = 'No records to display.';
        _this.allowFiltering = true;
        _this.allowSorting = true;
        _this.multiple = false;
        _this.loading = false;
        _this.searchValue = '';
        _this.multiSortMeta = [];
        _this.skip = 0;
        _this.top = 10;
        _this.filter = '';
        _this.orderBy = '';
        return _this;
    }
    Object.defineProperty(LookupPopupFormFieldComponent.prototype, "pageSize", {
        get: function () {
            return this.top;
        },
        set: function (value) {
            this.top = value;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LookupPopupFormFieldComponent.prototype, "options", {
        get: function () {
            var _this = this;
            var formControl = this.parent.form.controls[this.property];
            if (this.data && formControl) {
                var current_1 = formControl.value;
                if (!this.multiple) {
                    var item = this.valueProperty
                        ? this.data.find(function (i) { return i[_this.valueProperty] == current_1; })
                        : this.data.find(function (i) { return i == current_1; });
                    if (item) {
                        return [this.optionByItem(item)];
                    }
                    else if (this.value) {
                        return [this.optionByItem(this.value)];
                    }
                }
                else if (this.multiple) {
                    var items = this.selection || [];
                    var opts = items.map(function (item) { return _this.optionByItem(item); });
                    if (opts.length > 1) {
                        var labels = items.map(function (item) { return get(item, _this.textProperty); });
                        opts[0].label = [labels[0], labels[1]].join(', ') + ", ...";
                    }
                    return opts;
                }
            }
            return [];
        },
        enumerable: true,
        configurable: true
    });
    LookupPopupFormFieldComponent.prototype.onToggle = function (op, event) {
        var formControl = this.parent.form.controls[this.property];
        if (formControl && formControl.disabled != true) {
            op.toggle(event);
        }
    };
    LookupPopupFormFieldComponent.prototype.optionByItem = function (item) {
        return {
            label: this.textProperty ? get(item, this.textProperty) : item,
            value: this.valueProperty ? get(item, this.valueProperty) : item
        };
    };
    LookupPopupFormFieldComponent.prototype.updateValue = function () {
        var _this = this;
        if (this.selection) {
            var newValue = this.multiple
                ? this.selection.map(function (item) { return get(item, _this.valueProperty); })
                : get(this.selection, this.valueProperty);
            this.parent.form.controls[this.property].setValue(newValue, {
                emitEvent: true
            });
        }
    };
    LookupPopupFormFieldComponent.prototype.onSearch = function () {
        this.filter =
            this.searchValue && this.textProperty
                ? "contains(" + this.textProperty + ", '" + this.searchValue + "')"
                : '';
        this.callLoadData();
    };
    LookupPopupFormFieldComponent.prototype.onSearchKeyPress = function (event) {
        if (event.keyCode === 13) {
            this.onSearch();
        }
    };
    LookupPopupFormFieldComponent.prototype.onRowSelect = function (op, dropdown, event) {
        if (!this.multiple) {
            op.toggle(event, dropdown);
            this.updateValue();
        }
    };
    LookupPopupFormFieldComponent.prototype.onClose = function (op, dropdown, event) {
        op.toggle(event, dropdown);
        this.updateValue();
    };
    LookupPopupFormFieldComponent.prototype.onPage = function (event) {
        if (this.skip != event.first || this.top != event.rows) {
            this.skip = event.first;
            this.top = event.rows;
            this.callLoadData();
        }
    };
    LookupPopupFormFieldComponent.prototype.onSort = function (event) {
        if (this.allowSorting && event.field) {
            var field = event.multisortmeta.find(function (i) { return i && i.field == event.field; });
            event.multisortmeta.splice(event.multisortmeta.indexOf(field), 1);
            event.multisortmeta.push({ field: event.field, order: event.order });
        }
        this.orderBy = this.allowSorting
            ? grid_1.orderByToString(event.multisortmeta)
            : '';
        this.callLoadData();
    };
    LookupPopupFormFieldComponent.prototype.callLoadData = function () {
        this.loading = true;
        this.parent.loadData.next({
            property: this.property,
            skip: this.skip,
            top: this.top,
            orderby: this.orderBy,
            filter: this.filter
        });
    };
    LookupPopupFormFieldComponent.prototype.ngOnChanges = function (changes) {
        if (changes.hasOwnProperty('data')) {
            this.loading = false;
        }
        if (changes.hasOwnProperty('valueProperty') ||
            changes.hasOwnProperty('textProperty')) {
            this.columns =
                this.valueProperty && this.textProperty
                    ? [
                        {
                            property: this.valueProperty,
                            title: this.valueProperty
                        },
                        {
                            property: this.textProperty,
                            title: this.textProperty
                        }
                    ]
                    : [];
        }
    };
    LookupPopupFormFieldComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-lookuppopup-form-field',
                    template: "\n    <p-overlayPanel appendTo=\"body\" #op>\n      <div class=\"ui-lookup-panel\">\n        <div class=\"ui-lookup-search\">\n          <input #search (keypress)=\"onSearchKeyPress($event)\" *ngIf=\"allowFiltering\" [(ngModel)]=\"searchValue\" placeholder=\"Search...\">\n          <rz-button *ngIf=\"allowFiltering\" icon=\"search\" (click)=\"onSearch()\"></rz-button>\n        </div>\n        <p-dataTable\n          [selectionMode]=\"multiple? 'multiple' : 'single'\"\n          [(selection)]=\"selection\"\n          scrollable=\"true\"\n          scrollHeight=\"200px\"\n          scrollWidth=\"400px\"\n          responsive=\"true\"\n          alwaysShowPaginator=\"false\"\n          [lazy]=\"parent.loadData.observers.length > 0\"\n          [loading]=\"isLoading || (parent.loadData.observers.length > 0 && loading)\"\n          [value]=\"data\"\n          [totalRecords]=\"count\"\n          [emptyMessage]=\"emptyText\"\n          (onPage)=\"onPage($event)\"\n          (onSort)=\"onSort($event)\"\n          (onRowSelect)=\"onRowSelect(op, dropdown, $event)\"\n          [sortMode]=\"allowSorting ? 'single' : null\"\n          [rows]=\"pageSize\"\n          [multiSortMeta]=\"multiSortMeta\"\n          paginator=\"true\"\n          #dataTable>\n          <p-column *ngFor=\"let column of columns; let index = index\" [field]=\"column.property\" [header]=\"column.title\"\n            [sortable]=\"allowSorting ? (parent.loadData.observers.length > 0 ? 'custom' : true): false\"></p-column>\n        </p-dataTable>\n        <div *ngIf=\"multiple\" class=\"ui-lookup-select\">\n          <rz-button text=\"Select\" (click)=\"onClose(op, dropdown, $event)\"></rz-button>\n        </div>\n      </div>\n    </p-overlayPanel>\n    <div #dropdown class=\"ui-lookup\" (click)=\"onToggle(op, $event)\">\n      <p-dropdown [options]=\"options\" *ngIf=\"data\" [inputId]=\"property\" [placeholder]=\"placeholder\" readonly=\"true\"\n        [autoWidth]=\"false\" [formControl]=\"parent.form.controls[property]\">\n      </p-dropdown>\n    </div>\n    <div class=\"md-inputfield\"\n      *ngIf=\"(parent.form.controls[property].touched || parent.submitted) && parent.form.controls[property].errors?.required\">\n      <div class=\"ui-message ui-messages-error ui-corner-all\">\n        {{ title }} {{ requiredText }}\n      </div>\n    </div>\n  ",
                    providers: [
                        {
                            provide: form_field_1.FormFieldComponent,
                            useExisting: core_1.forwardRef(function () { return LookupPopupFormFieldComponent; })
                        }
                    ]
                },] },
    ];
    /** @nocollapse */
    LookupPopupFormFieldComponent.ctorParameters = function () { return []; };
    LookupPopupFormFieldComponent.propDecorators = {
        "emptyText": [{ type: core_1.Input },],
        "textProperty": [{ type: core_1.Input },],
        "valueProperty": [{ type: core_1.Input },],
        "placeholder": [{ type: core_1.Input },],
        "data": [{ type: core_1.Input },],
        "value": [{ type: core_1.Input },],
        "count": [{ type: core_1.Input },],
        "allowFiltering": [{ type: core_1.Input },],
        "allowSorting": [{ type: core_1.Input },],
        "multiple": [{ type: core_1.Input },],
        "isLoading": [{ type: core_1.Input },],
        "pageSize": [{ type: core_1.Input },],
    };
    return LookupPopupFormFieldComponent;
}(form_field_1.FormFieldComponent));
exports.LookupPopupFormFieldComponent = LookupPopupFormFieldComponent;
