// @radzen/angular/dist/notification.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var Subject_1 = require("rxjs/Subject");
var growl_1 = require("primeng/growl");
var NotificationService = /** @class */ (function () {
    function NotificationService(componentFactoryResolver, applicationRef, injector) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.applicationRef = applicationRef;
        this.injector = injector;
        this.messages = [];
    }
    NotificationService.prototype.notify = function (message) {
        var _this = this;
        this.init();
        this.messages = this.messages.concat(message);
        this.componentRef.instance.value = this.messages;
        var subject = new Subject_1.Subject();
        var subscription = this.componentRef.instance.onClose.subscribe(function (event) {
            if (event.message == message) {
                _this.messages = _this.messages.filter(function (message) { return message !== event.message; });
                subject.next();
                subject.complete();
                subscription.unsubscribe();
            }
        });
        return subject;
    };
    NotificationService.prototype.init = function () {
        if (this.componentRef) {
            return;
        }
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(growl_1.Growl);
        this.componentRef = componentFactory.create(this.injector);
        this.componentRef.instance.value = this.messages;
        this.applicationRef.attachView(this.componentRef.hostView);
        document.body.appendChild(this.componentRef.hostView.rootNodes[0]);
    };
    NotificationService.decorators = [
        { type: core_1.Injectable },
    ];
    /** @nocollapse */
    NotificationService.ctorParameters = function () { return [
        { type: core_1.ComponentFactoryResolver, },
        { type: core_1.ApplicationRef, },
        { type: core_1.Injector, },
    ]; };
    return NotificationService;
}());
exports.NotificationService = NotificationService;
var NotificationModule = /** @class */ (function () {
    function NotificationModule() {
    }
    NotificationModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, growl_1.GrowlModule],
                    providers: [NotificationService],
                    entryComponents: [growl_1.Growl]
                },] },
    ];
    /** @nocollapse */
    NotificationModule.ctorParameters = function () { return []; };
    return NotificationModule;
}());
exports.NotificationModule = NotificationModule;
