// @radzen/angular/dist/number-form-field.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var form_field_1 = require("./form-field");
var NumberFormFieldComponent = /** @class */ (function (_super) {
    __extends(NumberFormFieldComponent, _super);
    function NumberFormFieldComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.placeholder = '';
        _this.autoComplete = true;
        _this.minText = 'cannot be less than';
        _this.maxText = 'cannot be more than';
        return _this;
    }
    NumberFormFieldComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-number-form-field',
                    template: "\n    <rz-numeric\n      [formControl]=\"parent.form.controls[property]\"\n      [placeholder]=\"placeholder\"\n      [autoComplete]=\"autoComplete\"\n      [min]=\"min\"\n      [max]=\"max\"\n      [name]=\"property\"></rz-numeric>\n    <div class=\"md-inputfield\"\n      *ngIf=\"(parent.form.controls[property].touched || parent.submitted) && parent.form.controls[property].errors?.required\">\n      <div class=\"ui-message ui-messages-error ui-corner-all\">\n        {{ title }} {{ requiredText }}\n      </div>\n    </div>\n    <div class=\"md-inputfield\"\n      *ngIf=\"(parent.form.controls[property].touched || parent.submitted) && parent.form.controls[property].errors?.min\">\n      <div class=\"ui-message ui-messages-error ui-corner-all\">\n        {{ title }} {{ minText }} {{min}}\n      </div>\n    </div>\n    <div class=\"md-inputfield\"\n      *ngIf=\"(parent.form.controls[property].touched || parent.submitted) && parent.form.controls[property].errors?.max\">\n      <div class=\"ui-message ui-messages-error ui-corner-all\">\n        {{ title }} {{ maxText }} {{max}}\n      </div>\n    </div>\n  ",
                    providers: [
                        {
                            provide: form_field_1.FormFieldComponent,
                            useExisting: core_1.forwardRef(function () { return NumberFormFieldComponent; })
                        }
                    ]
                },] },
    ];
    /** @nocollapse */
    NumberFormFieldComponent.ctorParameters = function () { return []; };
    NumberFormFieldComponent.propDecorators = {
        "placeholder": [{ type: core_1.Input },],
        "min": [{ type: core_1.Input },],
        "max": [{ type: core_1.Input },],
        "autoComplete": [{ type: core_1.Input },],
        "minText": [{ type: core_1.Input },],
        "maxText": [{ type: core_1.Input },],
    };
    return NumberFormFieldComponent;
}(form_field_1.FormFieldComponent));
exports.NumberFormFieldComponent = NumberFormFieldComponent;
