// @radzen/angular/dist/numeric-range-validator.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var forms_1 = require("@angular/forms");
var validator_1 = require("./validator");
var NumericRangeValidatorComponent = /** @class */ (function (_super) {
    __extends(NumericRangeValidatorComponent, _super);
    function NumericRangeValidatorComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.text = "Not in the valid range";
        _this.min = 0;
        return _this;
    }
    Object.defineProperty(NumericRangeValidatorComponent.prototype, "validate", {
        get: function () {
            var validator = null;
            if (!isNaN(this.min)) {
                validator = forms_1.Validators.min(this.min);
            }
            if (!isNaN(this.max)) {
                var max = forms_1.Validators.max(this.max);
                if (validator) {
                    validator = forms_1.Validators.compose([validator, max]);
                }
                else {
                    validator = max;
                }
            }
            return validator;
        },
        enumerable: true,
        configurable: true
    });
    NumericRangeValidatorComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-numeric-range-validator',
                    template: "\n    <div [ngStyle]=\"style\" class=\"ui-message ui-messages-error\"\n      *ngIf=\"token || ((control?.errors?.min || control?.errors?.max) && (control?.touched || control?.dirty || form?.ngForm?.submitted))\">\n        {{ text }}\n    </div>\n  "
                },] },
    ];
    /** @nocollapse */
    NumericRangeValidatorComponent.ctorParameters = function () { return []; };
    NumericRangeValidatorComponent.propDecorators = {
        "text": [{ type: core_1.Input },],
        "min": [{ type: core_1.Input },],
        "max": [{ type: core_1.Input },],
    };
    return NumericRangeValidatorComponent;
}(validator_1.ValidatorComponent));
exports.NumericRangeValidatorComponent = NumericRangeValidatorComponent;
var NumericRangeValidatorModule = /** @class */ (function () {
    function NumericRangeValidatorModule() {
    }
    NumericRangeValidatorModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule],
                    declarations: [NumericRangeValidatorComponent],
                    exports: [NumericRangeValidatorComponent]
                },] },
    ];
    /** @nocollapse */
    NumericRangeValidatorModule.ctorParameters = function () { return []; };
    return NumericRangeValidatorModule;
}());
exports.NumericRangeValidatorModule = NumericRangeValidatorModule;
