// @radzen/angular/dist/panel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var panel_1 = require("primeng/panel");
var PanelHeaderComponent = /** @class */ (function () {
    function PanelHeaderComponent() {
    }
    PanelHeaderComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-panel-header',
                    template: "\n    <ng-content></ng-content>\n  "
                },] },
    ];
    /** @nocollapse */
    PanelHeaderComponent.ctorParameters = function () { return []; };
    return PanelHeaderComponent;
}());
exports.PanelHeaderComponent = PanelHeaderComponent;
var PanelFooterComponent = /** @class */ (function () {
    function PanelFooterComponent() {
    }
    PanelFooterComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-panel-footer',
                    template: "\n    <ng-content></ng-content>\n  "
                },] },
    ];
    /** @nocollapse */
    PanelFooterComponent.ctorParameters = function () { return []; };
    return PanelFooterComponent;
}());
exports.PanelFooterComponent = PanelFooterComponent;
var PanelComponent = /** @class */ (function () {
    function PanelComponent() {
        this.style = {};
        this.allowCollapse = false;
        this.expand = new core_1.EventEmitter();
        this.collapse = new core_1.EventEmitter();
    }
    PanelComponent.prototype.onToggle = function (event) {
        if (event.collapsed) {
            this.collapse.next();
        }
        else {
            this.expand.next();
        }
    };
    PanelComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-panel',
                    template: "<p-panel (onAfterToggle)=\"onToggle($event)\" [style]=\"style\" [toggleable]=\"allowCollapse\" [header]=\"icon ? null : text\">\n    <p-header *ngIf=\"icon\">\n      <i class=\"material-icons\">{{icon}}</i><span class=\"material-icons-label\">{{text}}</span>\n    </p-header>\n    <p-header>\n      <ng-content select=\"rz-panel-header\"></ng-content>\n    </p-header>\n    <p-footer>\n      <ng-content select=\"rz-panel-footer\"></ng-content>\n    </p-footer>\n    <ng-content></ng-content>\n  </p-panel>"
                },] },
    ];
    /** @nocollapse */
    PanelComponent.ctorParameters = function () { return []; };
    PanelComponent.propDecorators = {
        "text": [{ type: core_1.Input },],
        "icon": [{ type: core_1.Input },],
        "style": [{ type: core_1.Input },],
        "allowCollapse": [{ type: core_1.Input },],
        "expand": [{ type: core_1.Output },],
        "collapse": [{ type: core_1.Output },],
    };
    return PanelComponent;
}());
exports.PanelComponent = PanelComponent;
var PanelModule = /** @class */ (function () {
    function PanelModule() {
    }
    PanelModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, panel_1.PanelModule],
                    declarations: [PanelComponent, PanelFooterComponent, PanelHeaderComponent],
                    exports: [PanelComponent, PanelFooterComponent, PanelHeaderComponent]
                },] },
    ];
    /** @nocollapse */
    PanelModule.ctorParameters = function () { return []; };
    return PanelModule;
}());
exports.PanelModule = PanelModule;
