// @radzen/angular/dist/password.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var forms_1 = require("@angular/forms");
var form_component_1 = require("./form-component");
var PasswordComponent = /** @class */ (function (_super) {
    __extends(PasswordComponent, _super);
    function PasswordComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.style = {};
        _this.readOnly = false;
        _this.autoComplete = true;
        _this.placeholder = '';
        return _this;
    }
    PasswordComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-password',
                    template: "\n    <input\n      type=\"password\"\n      class=\"textbox\"\n      [ngStyle]=\"style\"\n      [readonly]=\"readOnly\"\n      [disabled]=\"disabled\"\n      [id]=\"name\"\n      [placeholder]=\"placeholder\"\n      [autocomplete]=\"autoComplete ? 'on' : 'off'\"\n      [(ngModel)]=\"value\"\n      (blur)=\"onTouched()\"\n      (change)=\"$event.stopPropagation()\">\n  ",
                    providers: [
                        {
                            provide: forms_1.NG_VALUE_ACCESSOR,
                            useExisting: core_1.forwardRef(function () { return PasswordComponent; }),
                            multi: true
                        },
                        {
                            provide: form_component_1.FormComponent,
                            useExisting: core_1.forwardRef(function () { return PasswordComponent; })
                        }
                    ]
                },] },
    ];
    /** @nocollapse */
    PasswordComponent.ctorParameters = function () { return []; };
    PasswordComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "readOnly": [{ type: core_1.Input },],
        "autoComplete": [{ type: core_1.Input },],
        "placeholder": [{ type: core_1.Input },],
    };
    return PasswordComponent;
}(form_component_1.FormComponent));
exports.PasswordComponent = PasswordComponent;
var PasswordModule = /** @class */ (function () {
    function PasswordModule() {
    }
    PasswordModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, forms_1.FormsModule],
                    declarations: [PasswordComponent],
                    exports: [PasswordComponent]
                },] },
    ];
    /** @nocollapse */
    PasswordModule.ctorParameters = function () { return []; };
    return PasswordModule;
}());
exports.PasswordModule = PasswordModule;
