// @radzen/angular/dist/picklist.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var picklist_1 = require("primeng/picklist");
var PickListComponent = /** @class */ (function () {
    function PickListComponent() {
        this.style = {};
        this.source = [];
        this.target = [];
        this.showSourceFilter = true;
        this.showTargetFilter = true;
        this.dragdrop = false;
        this.showSourceControls = true;
        this.showTargetControls = true;
        this.visible = true;
        this.disabled = false;
        this.defaultItemText = 'Template for index';
        this.moveToTarget = new core_1.EventEmitter();
        this.moveToSource = new core_1.EventEmitter();
        this.moveAllToTarget = new core_1.EventEmitter();
        this.moveAllToSource = new core_1.EventEmitter();
        this.sourceReorder = new core_1.EventEmitter();
        this.targetReorder = new core_1.EventEmitter();
        this.sourceSelect = new core_1.EventEmitter();
        this.targetSelect = new core_1.EventEmitter();
    }
    PickListComponent.prototype.onMoveToTarget = function (event) {
        this.moveToTarget.next(event);
    };
    PickListComponent.prototype.onMoveToSource = function (event) {
        this.moveToSource.next(event);
    };
    PickListComponent.prototype.onMoveAllToTarget = function (event) {
        this.moveAllToTarget.next(event);
    };
    PickListComponent.prototype.onMoveAllToSource = function (event) {
        this.moveAllToSource.next(event);
    };
    PickListComponent.prototype.onSourceReorder = function (event) {
        this.sourceReorder.next(event);
    };
    PickListComponent.prototype.onTargetReorder = function (event) {
        this.targetReorder.next(event);
    };
    PickListComponent.prototype.onSourceSelect = function (event) {
        this.sourceSelect.next(event);
    };
    PickListComponent.prototype.onTargetSelect = function (event) {
        this.targetSelect.next(event);
    };
    PickListComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-picklist',
                    template: "<p-pickList [styleClass]=\"disabled ? 'ui-state-disabled':null\" [style]=\"style\" [disabled]=\"disabled\" [source]=\"source\" [target]=\"target\" [responsive]=\"true\"\n    [sourceHeader]=\"sourceHeader\" [targetHeader]=\"targetHeader\"\n    [filterBy]=\"filterBy\" [showSourceFilter]=\"showSourceFilter\" [showTargetFilter]=\"showTargetFilter\"\n    [dragdrop]=\"dragdrop\" [sourceFilterPlaceholder]=\"sourceFilterPlaceholder\" [targetFilterPlaceholder]=\"targetFilterPlaceholder\"\n    [showSourceControls]=\"showSourceControls\" [showTargetControls]=\"showTargetControls\"\n    (onMoveToTarget)=\"onMoveToTarget($event)\"\n    (onMoveToSource)=\"onMoveToSource($event)\"\n    (onMoveAllToTarget)=\"onMoveAllToTarget($event)\"\n    (onMoveAllToSource)=\"onMoveAllToSource($event)\"\n    (onSourceReorder)=\"onSourceReorder($event)\"\n    (onTargetReorder)=\"onTargetReorder($event)\"\n    (onSourceSelect)=\"onSourceSelect($event)\"\n    (onTargetSelect)=\"onTargetSelect($event)\">\n      <ng-template *ngIf=\"template\" pTemplate=\"item\" let-data>\n        <ng-template [ngTemplateOutlet]=\"template\" [ngTemplateOutletContext]=\"{ $implicit: data }\"></ng-template>\n      </ng-template>\n      <ng-template *ngIf=\"!template\" pTemplate=\"item\" let-i=\"index\">\n        {{ defaultItemText }}: {{i}}\n      </ng-template>\n  </p-pickList>"
                },] },
    ];
    /** @nocollapse */
    PickListComponent.ctorParameters = function () { return []; };
    PickListComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "source": [{ type: core_1.Input },],
        "target": [{ type: core_1.Input },],
        "sourceHeader": [{ type: core_1.Input },],
        "targetHeader": [{ type: core_1.Input },],
        "sourceFilterPlaceholder": [{ type: core_1.Input },],
        "targetFilterPlaceholder": [{ type: core_1.Input },],
        "filterBy": [{ type: core_1.Input },],
        "showSourceFilter": [{ type: core_1.Input },],
        "showTargetFilter": [{ type: core_1.Input },],
        "dragdrop": [{ type: core_1.Input },],
        "showSourceControls": [{ type: core_1.Input },],
        "showTargetControls": [{ type: core_1.Input },],
        "visible": [{ type: core_1.Input },],
        "disabled": [{ type: core_1.Input },],
        "defaultItemText": [{ type: core_1.Input },],
        "template": [{ type: core_1.ContentChild, args: [core_1.TemplateRef,] },],
        "moveToTarget": [{ type: core_1.Output },],
        "moveToSource": [{ type: core_1.Output },],
        "moveAllToTarget": [{ type: core_1.Output },],
        "moveAllToSource": [{ type: core_1.Output },],
        "sourceReorder": [{ type: core_1.Output },],
        "targetReorder": [{ type: core_1.Output },],
        "sourceSelect": [{ type: core_1.Output },],
        "targetSelect": [{ type: core_1.Output },],
    };
    return PickListComponent;
}());
exports.PickListComponent = PickListComponent;
var PickListModule = /** @class */ (function () {
    function PickListModule() {
    }
    PickListModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, picklist_1.PickListModule],
                    declarations: [PickListComponent],
                    exports: [PickListComponent]
                },] },
    ];
    /** @nocollapse */
    PickListModule.ctorParameters = function () { return []; };
    return PickListModule;
}());
exports.PickListModule = PickListModule;
