// @radzen/angular/dist/profilemenu.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var Subject_1 = require("rxjs/Subject");
var menu_1 = require("./menu");
var navigation_1 = require("./navigation");
var ProfileMenuItemComponent = /** @class */ (function (_super) {
    __extends(ProfileMenuItemComponent, _super);
    function ProfileMenuItemComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ProfileMenuItemComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-profilemenu-item',
                    template: ""
                },] },
    ];
    /** @nocollapse */
    ProfileMenuItemComponent.ctorParameters = function () { return []; };
    ProfileMenuItemComponent.propDecorators = {
        "items": [{ type: core_1.ContentChildren, args: [ProfileMenuItemComponent,] },],
    };
    return ProfileMenuItemComponent;
}(navigation_1.NavigationItemComponent));
exports.ProfileMenuItemComponent = ProfileMenuItemComponent;
var ProfileMenuTemplateDirective = /** @class */ (function () {
    function ProfileMenuTemplateDirective(template) {
        this.template = template;
    }
    ProfileMenuTemplateDirective.decorators = [
        { type: core_1.Directive, args: [{
                    selector: '[profilemenuTemplate]'
                },] },
    ];
    /** @nocollapse */
    ProfileMenuTemplateDirective.ctorParameters = function () { return [
        { type: core_1.TemplateRef, },
    ]; };
    return ProfileMenuTemplateDirective;
}());
exports.ProfileMenuTemplateDirective = ProfileMenuTemplateDirective;
var ProfileMenuComponent = /** @class */ (function () {
    function ProfileMenuComponent() {
        var _this = this;
        this.style = {};
        this.click = new core_1.EventEmitter();
        this.ngUnsubscribe = new Subject_1.Subject();
        this.childrenToItems = function () {
            _this.items = navigation_1.childrenToItems(_this.children.toArray());
        };
    }
    ProfileMenuComponent.prototype.ngAfterContentInit = function () {
        if (this.children) {
            this.childrenToItems();
            this.children.changes.subscribe(this.childrenToItems);
        }
    };
    ProfileMenuComponent.prototype.ngOnDestroy = function () {
        this.ngUnsubscribe.next();
        this.ngUnsubscribe.complete();
    };
    ProfileMenuComponent.prototype.onClick = function (event) {
        this.click.next(event);
    };
    ProfileMenuComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-profilemenu',
                    template: "\n    <rz-menu [style]=\"style\" class=\"profile-menu\" (click)=\"onClick($event)\">\n      <rz-menu-item [template]=\"profileTemplate?.template\">\n        <rz-menu-item *ngFor=\"let item of items\" [text]=\"item.text\" [icon]=\"item.icon\" [path]=\"item.path\">\n        </rz-menu-item>\n      </rz-menu-item>\n    </rz-menu>\n  "
                },] },
    ];
    /** @nocollapse */
    ProfileMenuComponent.ctorParameters = function () { return []; };
    ProfileMenuComponent.propDecorators = {
        "items": [{ type: core_1.Input },],
        "style": [{ type: core_1.Input },],
        "click": [{ type: core_1.Output },],
        "children": [{ type: core_1.ContentChildren, args: [ProfileMenuItemComponent,] },],
        "profileTemplate": [{ type: core_1.ContentChild, args: [ProfileMenuTemplateDirective,] },],
    };
    return ProfileMenuComponent;
}());
exports.ProfileMenuComponent = ProfileMenuComponent;
var ProfileMenuModule = /** @class */ (function () {
    function ProfileMenuModule() {
    }
    ProfileMenuModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, menu_1.MenuModule],
                    declarations: [
                        ProfileMenuTemplateDirective,
                        ProfileMenuComponent,
                        ProfileMenuItemComponent
                    ],
                    exports: [
                        ProfileMenuTemplateDirective,
                        ProfileMenuComponent,
                        ProfileMenuItemComponent
                    ]
                },] },
    ];
    /** @nocollapse */
    ProfileMenuModule.ctorParameters = function () { return []; };
    return ProfileMenuModule;
}());
exports.ProfileMenuModule = ProfileMenuModule;
