// @radzen/angular/dist/progressbar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var progressbar_1 = require("primeng/progressbar");
var shared_1 = require("./shared");
var ProgressBarComponent = /** @class */ (function () {
    function ProgressBarComponent() {
        this.style = {};
        this.showValue = true;
        this.unit = '%';
        this.mode = 'determinate';
    }
    ProgressBarComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-progressbar',
                    template: "\n  <p-progressBar\n    [style]=\"style\"\n    [showValue]=\"showValue\"\n    [unit]=\"unit\"\n    [mode]=\"mode\"\n    [value]=\"value\">\n  </p-progressBar>\n  "
                },] },
    ];
    /** @nocollapse */
    ProgressBarComponent.ctorParameters = function () { return []; };
    ProgressBarComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "value": [{ type: core_1.Input },],
        "showValue": [{ type: core_1.Input },],
        "unit": [{ type: core_1.Input },],
        "mode": [{ type: core_1.Input },],
    };
    return ProgressBarComponent;
}());
exports.ProgressBarComponent = ProgressBarComponent;
var ProgressBarModule = /** @class */ (function () {
    function ProgressBarModule() {
    }
    ProgressBarModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, progressbar_1.ProgressBarModule, shared_1.SharedModule],
                    declarations: [ProgressBarComponent],
                    exports: [ProgressBarComponent]
                },] },
    ];
    /** @nocollapse */
    ProgressBarModule.ctorParameters = function () { return []; };
    return ProgressBarModule;
}());
exports.ProgressBarModule = ProgressBarModule;
