// @radzen/angular/dist/radiobuttonlist.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var forms_1 = require("@angular/forms");
var radiobutton_1 = require("primeng/radiobutton");
var form_component_1 = require("./form-component");
var options_component_1 = require("./options-component");
var RadioButtonListOptionComponent = /** @class */ (function () {
    function RadioButtonListOptionComponent() {
    }
    RadioButtonListOptionComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-radiobuttonlist-option',
                    template: ''
                },] },
    ];
    /** @nocollapse */
    RadioButtonListOptionComponent.ctorParameters = function () { return []; };
    RadioButtonListOptionComponent.propDecorators = {
        "text": [{ type: core_1.Input },],
        "value": [{ type: core_1.Input },],
    };
    return RadioButtonListOptionComponent;
}());
exports.RadioButtonListOptionComponent = RadioButtonListOptionComponent;
var RadioButtonListComponent = /** @class */ (function (_super) {
    __extends(RadioButtonListComponent, _super);
    function RadioButtonListComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.style = {};
        _this.orientation = 'horizontal';
        return _this;
    }
    RadioButtonListComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-radiobuttonlist',
                    template: "\n    <div\n      [ngClass]=\"{\n        'radio-button-list-horizontal': orientation == 'horizontal',\n        'radio-button-list-vertical': orientation == 'vertical'\n      }\"\n      [id]=\"name\"\n      [ngStyle]=\"style\"\n    >\n      <p-radioButton\n        *ngFor=\"let option of options\"\n        [value]=\"option.value\"\n        [label]=\"option.label\"\n        [(ngModel)]=\"value\"\n        [name]=\"name\"\n        [disabled]=\"disabled\"\n      >\n      </p-radioButton>\n    </div>\n  ",
                    providers: [
                        {
                            provide: forms_1.NG_VALUE_ACCESSOR,
                            useExisting: core_1.forwardRef(function () { return RadioButtonListComponent; }),
                            multi: true
                        },
                        {
                            provide: form_component_1.FormComponent,
                            useExisting: core_1.forwardRef(function () { return RadioButtonListComponent; })
                        }
                    ]
                },] },
    ];
    /** @nocollapse */
    RadioButtonListComponent.ctorParameters = function () { return []; };
    RadioButtonListComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "orientation": [{ type: core_1.Input },],
        "children": [{ type: core_1.ContentChildren, args: [RadioButtonListOptionComponent,] },],
    };
    return RadioButtonListComponent;
}(options_component_1.OptionsComponent));
exports.RadioButtonListComponent = RadioButtonListComponent;
var RadioButtonListModule = /** @class */ (function () {
    function RadioButtonListModule() {
    }
    RadioButtonListModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, radiobutton_1.RadioButtonModule, forms_1.FormsModule],
                    declarations: [RadioButtonListComponent, RadioButtonListOptionComponent],
                    exports: [RadioButtonListComponent, RadioButtonListOptionComponent]
                },] },
    ];
    /** @nocollapse */
    RadioButtonListModule.ctorParameters = function () { return []; };
    return RadioButtonListModule;
}());
exports.RadioButtonListModule = RadioButtonListModule;
