// @radzen/angular/dist/rating.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var forms_1 = require("@angular/forms");
var rating_1 = require("primeng/rating");
var form_component_1 = require("./form-component");
var RatingComponent = /** @class */ (function (_super) {
    __extends(RatingComponent, _super);
    function RatingComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.stars = 5;
        return _this;
    }
    RatingComponent.prototype.onRate = function (event) {
        this.change.emit(event.value);
    };
    RatingComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-rating',
                    template: "<p-rating [cancel]=\"true\" [disabled]=\"disabled\" [stars]=\"stars\" [(ngModel)]=\"value\"></p-rating>",
                    providers: [
                        {
                            provide: forms_1.NG_VALUE_ACCESSOR,
                            useExisting: core_1.forwardRef(function () { return RatingComponent; }),
                            multi: true
                        },
                        {
                            provide: form_component_1.FormComponent,
                            useExisting: core_1.forwardRef(function () { return RatingComponent; })
                        }
                    ]
                },] },
    ];
    /** @nocollapse */
    RatingComponent.ctorParameters = function () { return []; };
    RatingComponent.propDecorators = {
        "stars": [{ type: core_1.Input },],
    };
    return RatingComponent;
}(form_component_1.FormComponent));
exports.RatingComponent = RatingComponent;
var RatingModule = /** @class */ (function () {
    function RatingModule() {
    }
    RatingModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [forms_1.FormsModule, rating_1.RatingModule],
                    declarations: [RatingComponent],
                    exports: [RatingComponent]
                },] },
    ];
    /** @nocollapse */
    RatingModule.ctorParameters = function () { return []; };
    return RatingModule;
}());
exports.RatingModule = RatingModule;
