// @radzen/angular/dist/required-validator.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var forms_1 = require("@angular/forms");
var validator_1 = require("./validator");
var RequiredValidatorComponent = /** @class */ (function (_super) {
    __extends(RequiredValidatorComponent, _super);
    function RequiredValidatorComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.text = 'Required';
        _this.validate = forms_1.Validators.required;
        return _this;
    }
    RequiredValidatorComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-required-validator',
                    template: "\n    <div [ngStyle]=\"style\" class=\"ui-message ui-messages-error\"\n      *ngIf=\"token || (control?.errors?.required && (control?.touched || control?.dirty || form?.ngForm?.submitted))\">\n        {{ text }}\n    </div>\n  "
                },] },
    ];
    /** @nocollapse */
    RequiredValidatorComponent.ctorParameters = function () { return []; };
    RequiredValidatorComponent.propDecorators = {
        "text": [{ type: core_1.Input },],
    };
    return RequiredValidatorComponent;
}(validator_1.ValidatorComponent));
exports.RequiredValidatorComponent = RequiredValidatorComponent;
var RequiredValidatorModule = /** @class */ (function () {
    function RequiredValidatorModule() {
    }
    RequiredValidatorModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule],
                    declarations: [RequiredValidatorComponent],
                    exports: [RequiredValidatorComponent]
                },] },
    ];
    /** @nocollapse */
    RequiredValidatorModule.ctorParameters = function () { return []; };
    return RequiredValidatorModule;
}());
exports.RequiredValidatorModule = RequiredValidatorModule;
