// @radzen/angular/dist/sidebar-toggle.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var layout_1 = require("./layout");
var SidebarToggleComponent = /** @class */ (function () {
    function SidebarToggleComponent(layoutService) {
        this.layoutService = layoutService;
        this.style = {};
        this.icon = 'menu';
    }
    SidebarToggleComponent.prototype.onClick = function () {
        this.layoutService.toggleSidebar();
    };
    SidebarToggleComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-sidebar-toggle',
                    template: "\n  <button class=\"sidebar-toggle\" (click)=\"onClick()\" [ngStyle]=\"style\">\n    <i class=\"material-icons d-flex align-items-center\">{{icon}}</i>\n  </button>\n  "
                },] },
    ];
    /** @nocollapse */
    SidebarToggleComponent.ctorParameters = function () { return [
        { type: layout_1.LayoutService, },
    ]; };
    SidebarToggleComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "icon": [{ type: core_1.Input },],
    };
    return SidebarToggleComponent;
}());
exports.SidebarToggleComponent = SidebarToggleComponent;
var SidebarToggleModule = /** @class */ (function () {
    function SidebarToggleModule() {
    }
    SidebarToggleModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, layout_1.LayoutModule],
                    declarations: [SidebarToggleComponent],
                    exports: [SidebarToggleComponent]
                },] },
    ];
    /** @nocollapse */
    SidebarToggleModule.ctorParameters = function () { return []; };
    return SidebarToggleModule;
}());
exports.SidebarToggleModule = SidebarToggleModule;
