// @radzen/angular/dist/slider.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var forms_1 = require("@angular/forms");
var slider_1 = require("primeng/slider");
var form_component_1 = require("./form-component");
var SliderComponent = /** @class */ (function (_super) {
    __extends(SliderComponent, _super);
    function SliderComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.style = {};
        _this.step = 1;
        _this.min = 0;
        _this.max = 100;
        _this.range = false;
        return _this;
    }
    SliderComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-slider',
                    template: "\n    <p-slider\n      [step]=\"step\"\n      [min]=\"min\"\n      [max]=\"max\"\n      [style]=\"style\"\n      [disabled]=\"disabled\"\n      [range]=\"range\"\n      orientation=\"horizontal\"\n      [(ngModel)]=\"value\"\n      (onBlur)=\"onTouched()\"></p-slider>\n  ",
                    providers: [
                        {
                            provide: forms_1.NG_VALUE_ACCESSOR,
                            useExisting: core_1.forwardRef(function () { return SliderComponent; }),
                            multi: true
                        },
                        {
                            provide: form_component_1.FormComponent,
                            useExisting: core_1.forwardRef(function () { return SliderComponent; })
                        }
                    ]
                },] },
    ];
    /** @nocollapse */
    SliderComponent.ctorParameters = function () { return []; };
    SliderComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "step": [{ type: core_1.Input },],
        "min": [{ type: core_1.Input },],
        "max": [{ type: core_1.Input },],
        "range": [{ type: core_1.Input },],
    };
    return SliderComponent;
}(form_component_1.FormComponent));
exports.SliderComponent = SliderComponent;
var SliderModule = /** @class */ (function () {
    function SliderModule() {
    }
    SliderModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule, forms_1.FormsModule, slider_1.SliderModule],
                    declarations: [SliderComponent],
                    exports: [SliderComponent]
                },] },
    ];
    /** @nocollapse */
    SliderModule.ctorParameters = function () { return []; };
    return SliderModule;
}());
exports.SliderModule = SliderModule;
