// @radzen/angular/dist/steps.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var StepsItemComponent = /** @class */ (function () {
    function StepsItemComponent() {
        this.selected = false;
        this.disabled = false;
    }
    StepsItemComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-steps-item',
                    template: "\n    <div *ngIf=\"selected\" class=\"ui-widget-content\">\n      <ng-content></ng-content>\n    </div>\n  "
                },] },
    ];
    /** @nocollapse */
    StepsItemComponent.ctorParameters = function () { return []; };
    StepsItemComponent.propDecorators = {
        "text": [{ type: core_1.Input },],
        "selected": [{ type: core_1.Input },],
        "disabled": [{ type: core_1.Input },],
    };
    return StepsItemComponent;
}());
exports.StepsItemComponent = StepsItemComponent;
var StepsComponent = /** @class */ (function () {
    function StepsComponent() {
        this.style = {};
        this.nextText = 'Next';
        this.previousText = 'Previous';
        this.change = new core_1.EventEmitter();
        this.selection = 0;
    }
    Object.defineProperty(StepsComponent.prototype, "selectedIndex", {
        get: function () {
            return this.selection;
        },
        set: function (selectedIndex) {
            this.selection = selectedIndex;
            if (this.steps) {
                this.steps.forEach(function (step, index) {
                    step.selected = index === selectedIndex;
                });
            }
        },
        enumerable: true,
        configurable: true
    });
    StepsComponent.prototype.ngAfterContentInit = function () {
        var _this = this;
        this.createItems();
        this.subscription = this.steps.changes.subscribe(function () { return _this.createItems(); });
    };
    StepsComponent.prototype.ngOnDestroy = function () {
        this.subscription.unsubscribe();
    };
    StepsComponent.prototype.createItems = function () {
        var _this = this;
        this.steps.forEach(function (step, index) {
            step.selected = index === _this.selectedIndex;
        });
    };
    StepsComponent.prototype.canNavigateTo = function (index) {
        if (!this.steps) {
            return false;
        }
        var steps = this.steps.toArray();
        var step = steps[index];
        if (!step) {
            return false;
        }
        return step.disabled != true;
    };
    StepsComponent.prototype.select = function (index) {
        if (this.canNavigateTo(index)) {
            this.selectedIndex = index;
            this.change.next({
                index: this.selectedIndex
            });
        }
    };
    StepsComponent.prototype.onItemClick = function (event, index) {
        event.preventDefault();
        this.select(index);
    };
    StepsComponent.prototype.onPreviousClick = function (event) {
        event.preventDefault();
        this.select(this.selectedIndex - 1);
    };
    StepsComponent.prototype.onNextClick = function (event) {
        event.preventDefault();
        this.select(this.selectedIndex + 1);
    };
    StepsComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-steps',
                    template: "\n    <div stopChangePropagation [ngStyle]=\"style\">\n      <div class=\"ui-steps ui-widget ui-helper-clearfix\">\n        <ul role=\"tablist\">\n          <li\n            *ngFor=\"let item of steps; let i = index\"\n            class=\"ui-steps-item\"\n            [ngClass]=\"{\n              'ui-state-highlight ui-steps-current': item.selected,\n              'ui-state-default': !item.selected,\n              'ui-state-disabled': item.disabled\n            }\"\n          >\n            <a class=\"ui-menuitem-link\" (click)=\"onItemClick($event, i)\">\n              <span class=\"ui-steps-number\">{{ i + 1 }}</span>\n              <span class=\"ui-steps-title\">{{ item.text }}</span>\n            </a>\n          </li>\n        </ul>\n      </div>\n      <ng-content></ng-content>\n      <div class=\"ui-steps-buttons\">\n        <a class=\"ui-steps-prev\" (click)=\"onPreviousClick($event)\"\n          ><i class=\"material-icons\">navigate_before</i>{{ previousText }}</a\n        >\n        <a class=\"ui-steps-next\" (click)=\"onNextClick($event)\"\n          >{{ nextText }}<i class=\"material-icons\">navigate_next</i></a\n        >\n      </div>\n    </div>\n  "
                },] },
    ];
    /** @nocollapse */
    StepsComponent.ctorParameters = function () { return []; };
    StepsComponent.propDecorators = {
        "style": [{ type: core_1.Input },],
        "selectedIndex": [{ type: core_1.Input },],
        "nextText": [{ type: core_1.Input },],
        "previousText": [{ type: core_1.Input },],
        "change": [{ type: core_1.Output },],
        "steps": [{ type: core_1.ContentChildren, args: [StepsItemComponent,] },],
    };
    return StepsComponent;
}());
exports.StepsComponent = StepsComponent;
var StepsModule = /** @class */ (function () {
    function StepsModule() {
    }
    StepsModule.decorators = [
        { type: core_1.NgModule, args: [{
                    imports: [common_1.CommonModule],
                    declarations: [StepsComponent, StepsItemComponent],
                    exports: [StepsComponent, StepsItemComponent]
                },] },
    ];
    /** @nocollapse */
    StepsModule.ctorParameters = function () { return []; };
    return StepsModule;
}());
exports.StepsModule = StepsModule;
