// @radzen/angular/dist/string-form-field.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var form_field_1 = require("./form-field");
var StringFormFieldComponent = /** @class */ (function (_super) {
    __extends(StringFormFieldComponent, _super);
    function StringFormFieldComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.autoComplete = true;
        return _this;
    }
    StringFormFieldComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-string-form-field',
                    template: "\n      <input\n        class=\"textbox\"\n        [formControl]=\"parent.form.controls[property]\"\n        [attr.placeholder]=\"placeholder\"\n        [attr.autocomplete]=\"autoComplete ? 'on' : 'off'\"\n        [attr.maxlength]=\"maxLength\"\n        [attr.id]=\"property\">\n      <div class=\"md-inputfield\"\n        *ngIf=\"(parent.form.controls[property].touched || parent.submitted) && parent.form.controls[property].errors?.required\">\n        <div class=\"ui-message ui-messages-error ui-corner-all\">\n          {{ title }} {{ requiredText }}\n        </div>\n      </div>\n  ",
                    providers: [
                        {
                            provide: form_field_1.FormFieldComponent,
                            useExisting: core_1.forwardRef(function () { return StringFormFieldComponent; })
                        }
                    ]
                },] },
    ];
    /** @nocollapse */
    StringFormFieldComponent.ctorParameters = function () { return []; };
    StringFormFieldComponent.propDecorators = {
        "placeholder": [{ type: core_1.Input },],
        "autoComplete": [{ type: core_1.Input },],
        "maxLength": [{ type: core_1.Input },],
    };
    return StringFormFieldComponent;
}(form_field_1.FormFieldComponent));
exports.StringFormFieldComponent = StringFormFieldComponent;
