// @radzen/angular/dist/text-form-field.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var form_field_1 = require("./form-field");
var TextFormFieldComponent = /** @class */ (function (_super) {
    __extends(TextFormFieldComponent, _super);
    function TextFormFieldComponent() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.rows = 2;
        _this.cols = 20;
        return _this;
    }
    TextFormFieldComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: 'rz-text-form-field',
                    template: "\n      <textarea\n        class=\"textarea\"\n        [formControl]=\"parent.form.controls[property]\"\n        [attr.rows]=\"rows\"\n        [attr.cols]=\"cols\"\n        [attr.maxlength]=\"maxLength\"\n        [attr.placeholder]=\"placeholder\"\n        [attr.id]=\"property\"></textarea>\n      <div class=\"md-inputfield\"\n        *ngIf=\"(parent.form.controls[property].touched || parent.submitted) && parent.form.controls[property].errors?.required\">\n        <div class=\"ui-message ui-messages-error ui-corner-all\">\n          {{ title }} {{ requiredText }}\n        </div>\n      </div>\n  ",
                    providers: [
                        {
                            provide: form_field_1.FormFieldComponent,
                            useExisting: core_1.forwardRef(function () { return TextFormFieldComponent; })
                        }
                    ]
                },] },
    ];
    /** @nocollapse */
    TextFormFieldComponent.ctorParameters = function () { return []; };
    TextFormFieldComponent.propDecorators = {
        "placeholder": [{ type: core_1.Input },],
        "rows": [{ type: core_1.Input },],
        "cols": [{ type: core_1.Input },],
        "maxLength": [{ type: core_1.Input },],
    };
    return TextFormFieldComponent;
}(form_field_1.FormFieldComponent));
exports.TextFormFieldComponent = TextFormFieldComponent;
